src/back/glsl/mod.rs

```diff
@@ -405,18 +405,21 @@ impl<'a, W: Write> Writer<'a, W> {
             }
         }
 
+        let ep_info = self
+            .analysis
+            .get_entry_point(self.options.entry_point.0, &self.options.entry_point.1);
+
         // Write the globals
         //
         // We filter all globals that aren't used by the selected entry point as they might be
         // interfere with each other (i.e. two globals with the same location but different with
         // different classes)
-        for (handle, global) in self
-            .module
-            .global_variables
-            .iter()
-            .zip(&self.entry_point.function.global_usage)
-            .filter_map(|(global, usage)| Some(global).filter(|_| !usage.is_empty()))
-        {
+        for (handle, global) in self.module.global_variables.iter() {
+            let usage = ep_info[handle];
+            if usage.is_empty() {
```

Comment:
nit: inlining the usage seems better since it won't pollute the namespace and overall just looks better (in my opininion)